index.js

```diff
@@ -106,13 +106,17 @@ Sha512.WASM = wasm && wasm.buffer
 Sha512.WASM_SUPPORTED = typeof WebAssembly !== 'undefined'
 
 Sha512.ready = function (cb) {
-  if (!cb) cb = noop
   if (!wasm) return cb(new Error('WebAssembly not supported'))
 
+  if (cb) {
+    wasm.onload(cb)
+    return
+  }
+
   var p = new Promise(function (reject, resolve) {
```

Comment:
whoops, `reject` and `resolve` are actually flipped here, seems we introduced a bug 🤦